lib/src/priority_queue.dart

```diff
@@ -418,3 +455,41 @@ class HeapPriorityQueue<E> implements PriorityQueue<E> {
     _queue = newQueue;
   }
 }
+
+/// Implementation of [HeapPriorityQueue.unorderedElements].
+class _UnorderedElementsIterable<E> extends Iterable<E> {
+  final HeapPriorityQueue<E> _queue;
+  _UnorderedElementsIterable(this._queue);
+  @override
+  Iterator<E> get iterator => _UnorderedElementsIterator<E>(_queue);
+}
+
+class _UnorderedElementsIterator<E> implements Iterator<E> {
+  final HeapPriorityQueue<E> _queue;
+  final int _modificationCount;
```

Comment:
nit: `_initialModificationCount` or similar to indicate this is the count when this was created.

Reply:
Done.